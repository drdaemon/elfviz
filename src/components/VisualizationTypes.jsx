import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import VisualizationType from './VisualizationType';

const VisualizationTypes = ({ type, types, onChange }) => {
  return (
    <ul className="nav nav-pills">
      {types.map(typeObject => (
        <VisualizationType
          key={'type-' + typeObject.name}
          type={type}
          typeObject={typeObject}
          onChange={onChange.bind(this)}
        />
      ))}
    </ul>
  );
};

VisualizationTypes.propTypes = {
  type: PropTypes.string.isRequired,
  types: PropTypes.array.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default VisualizationTypes;
