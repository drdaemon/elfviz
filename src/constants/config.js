// height of the visualization chart (svg)
export const height = 450;

export const qrcode = 'data:image/jpeg;base64,iVBORw0KGgoAAAANSUhEUgAAAXIAAAFyAQMAAADS6sNKAAAABlBMVEX///8AAABVwtN+AAABpElEQVR4nO2a0Y2EMAxELVEAJdE6JVEAUpbYniQb0N3fnQbNfHiBPPOD4zjOmkmSJEmS9D8qqTOubKvG1vMaWjG2i+fl4+YCrqvlQqvZqvvuTh0Rz8nXz+5j22HNLPEsAkX8K/gaCh4UMfNN/Kv4QMNEPhf/Bt5/2lg3v+Z/8Qx8KoEvgzHxvPzo2dZri8rsZ4ln4D0UosAutQpzs0fRXZDexfPyUVYf+TgNprqZeG4+J/jagmKIDHuox8Qz8rlBLjCez9cZFU/IGwowj4dsdSGfOzA5iafiazz0AEgnhEc0rcVT89mgRgAso8m3iaflR8BaKOR+2Z4knogfp3qrwnpT8xYP4rn4+YApluqWz2sOEE/M+94Jq3S2N813yeWxHhPPxXvFZYYCe9hUNX6SeCa+tz2m/VTqWIp4Yn7yjKDo7c17v0s8FY/vXs+G+8z3Kgz9LvHMfNysJyY9zhMzHm75QTwZj//zYKpjP1W+Zr74F/BYqvOMEau5+DfwSZXmfrRR8cS8/6CBObiP6V08LZ9qQEFk+Ivu/U/xVLwkSZIkSX+pDyPI6zwnAxz3AAAAAElFTkSuQmCC';
