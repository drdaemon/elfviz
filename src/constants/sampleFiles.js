import elfAC97KernelDriver from '../data/elfAC97KernelDriver.json';
import elfOpenSSLPHPModule from '../data/elfOpenSSLPHPModule.json';
import elfOpenSSLPHPModuleAfterStrip from '../data/elfOpenSSLPHPModule-after-strip.json';
import elfSimpleCProgram from '../data/elfSimpleCProgram.json';

export default {
  elfAC97KernelDriver,
  elfOpenSSLPHPModule,
  elfOpenSSLPHPModuleAfterStrip,
  elfSimpleCProgram,
};
