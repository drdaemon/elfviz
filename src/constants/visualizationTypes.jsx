import React from 'react';

export const visualizationTypes = [
  {
    name: 'Icicles',
    icon: <i className="fa fa-2x fa-bar-chart fa-rotate-180" aria-hidden="true"></i>,
  },
  {
    name: 'Sunburst',
    icon: <i className="fa fa-2x fa-pie-chart" aria-hidden="true"></i>,
  },
];
