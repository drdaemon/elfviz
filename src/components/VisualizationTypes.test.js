import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import VisualizationTypes from './VisualizationTypes';
import { visualizationTypes } from '../constants/visualizationTypes';

describe('VisualizationTypes', () => {
  it('renders without crashing', () => {
    shallow(<VisualizationTypes type={''} types={visualizationTypes} onChange={() => {}} />);
  });

  it('should match a snapshot', () => {
    const wrapper = shallow(<VisualizationTypes type={'Icicle'} types={visualizationTypes} onChange={() => {}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
