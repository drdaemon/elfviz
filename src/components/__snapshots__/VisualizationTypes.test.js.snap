// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualizationTypes should match a snapshot 1`] = `
<ul
  className="nav nav-pills"
>
  <VisualizationType
    onChange={[Function]}
    type="Icicle"
    typeObject={
      Object {
        "icon": <i
          aria-hidden="true"
          className="fa fa-2x fa-bar-chart fa-rotate-180"
      />,
        "name": "Icicles",
      }
    }
  />
  <VisualizationType
    onChange={[Function]}
    type="Icicle"
    typeObject={
      Object {
        "icon": <i
          aria-hidden="true"
          className="fa fa-2x fa-pie-chart"
      />,
        "name": "Sunburst",
      }
    }
  />
</ul>
`;
